### PR TITLE
Update MXnet for diffsharp

### PR DESCRIPTION
**Pull request by dsyme**
Integrates dev --> mxnet

1. There is missing MXNet code in the backend for Conv1D, Copy and FlipT
2. Tests fail. I tried

       docker build -t diffsharp .
       docker run --rm diffsharp dotnet test

   but failure on both Windows and Linux due to missing "libmxnet".  

```
   System.DllNotFoundException : Unable to load shared library 'libmxnet' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: liblibmxnet: cannot open shared object file: No such file or directory
  Stack Trace:
     at MXNetSharp.Interop.CApi.MXNDArrayCreateEx(UInt32[] shape, UInt32 ndim, Int32 dev_type, Int32 dev_id, Int32 delay_alloc, Int32 dtype, IntPtr& out)
   at MXNetSharp.NDArray..ctor(IEnumerable`1 shape, Context context, FSharpOption`1 dtype, FSharpOption`1 delayAlloc)
```

### CODE REVIEW

File: src/DiffSharp.Backend.MXNet/RawTensorFloat32CPU.fs [hunk 1]
```diff
@@ -185,71 +188,81 @@ type RawTensorFloat32CPU(value: NDArray, shape:int[]) =
...
 
     override t1.AddTT0(t2) =
-        let t1value = t1.Value:?>NDArray
-        let t2value = t2.Value:?>NDArray  //TODO: check dim?
+        let t1value = t1.Value
+        let t2value = (t2 :?> RawTensorFloat32CPU).Value  //TODO: check dim?
         let result = t1value .+ t2value
         upcast RawTensorFloat32CPU(result, t1.Shape)
 
+    override t1.Conv1D(t2, stride, padding) = failwith "TBD - MXNet Conv1D"
```
Comment (dsyme) [inline src/DiffSharp.Backend.MXNet/RawTensorFloat32CPU.fs:246]: these are NYI

**Comment by kevmal** [comment]
I already had conv done and tested just not synced. merged it back in after merging this PR.. Added flip and copy just cause it only took a min. All tests pass, the libmxnet native dependency isn't included in the MXNetSharp nuget.

**Comment by dsyme** [comment]
Do you think we could include it in the NuGet?

**Comment by dsyme** [comment]
It would make using this waaay easy :-)

**Pull request merged**